<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Inferential-Statistics" data-toc-modified-id="Inferential-Statistics-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Inferential Statistics</a></span></li><li><span><a href="#Review-Set-Theory" data-toc-modified-id="Review-Set-Theory-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Review Set Theory</a></span></li><li><span><a href="#Classical-Probability" data-toc-modified-id="Classical-Probability-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Classical Probability</a></span><ul class="toc-item"><li><ul class="toc-item"><li><span><a href="#Terminology" data-toc-modified-id="Terminology-3.0.1"><span class="toc-item-num">3.0.1&nbsp;&nbsp;</span>Terminology</a></span></li></ul></li></ul></li><li><span><a href="#Handwritten-example---axiom-3" data-toc-modified-id="Handwritten-example---axiom-3-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Handwritten example - axiom 3</a></span><ul class="toc-item"><li><ul class="toc-item"><li><span><a href="#Handwritten-Example-2" data-toc-modified-id="Handwritten-Example-2-4.0.1"><span class="toc-item-num">4.0.1&nbsp;&nbsp;</span>Handwritten Example 2</a></span></li><li><span><a href="#Python-Example" data-toc-modified-id="Python-Example-4.0.2"><span class="toc-item-num">4.0.2&nbsp;&nbsp;</span>Python Example</a></span></li><li><span><a href="#Warm-up-Problem:-Die-Roll" data-toc-modified-id="Warm-up-Problem:-Die-Roll-4.0.3"><span class="toc-item-num">4.0.3&nbsp;&nbsp;</span>Warm-up Problem: Die Roll</a></span></li><li><span><a href="#Good.-Let's-do-a-few-more" data-toc-modified-id="Good.-Let's-do-a-few-more-4.0.4"><span class="toc-item-num">4.0.4&nbsp;&nbsp;</span>Good. Let's do a few more</a></span></li></ul></li></ul></li><li><span><a href="#Permutations" data-toc-modified-id="Permutations-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Permutations</a></span></li><li><span><a href="#Combinations" data-toc-modified-id="Combinations-6"><span class="toc-item-num">6&nbsp;&nbsp;</span>Combinations</a></span><ul class="toc-item"><li><ul class="toc-item"><li><ul class="toc-item"><li><span><a href="#Examples" data-toc-modified-id="Examples-6.0.0.1"><span class="toc-item-num">6.0.0.1&nbsp;&nbsp;</span>Examples</a></span></li><li><span><a href="#Card-Problems" data-toc-modified-id="Card-Problems-6.0.0.2"><span class="toc-item-num">6.0.0.2&nbsp;&nbsp;</span>Card Problems</a></span></li></ul></li></ul></li></ul></li></ul></div>

# Inferential Statistics 
1. What is it?
2. How is it different from descriptive statistics?
2. Why do we need it? 

![](http://www.z-table.com/uploads/2/1/7/9/21795380/205552900.jpg)

![](https://miro.medium.com/max/1164/1*WD8uN3s7_eo1peVehPZAVA.jpeg)

Earlier we looked at descriptive statistics: starting with a dataset and making various observations (overall shape, histogram, outliers, etc.) as well as calculations of quantities that can characterize the dataset as a whole (mean, median, mode, variance, standard deviation, quartiles, percentiles, etc.).

To make the move into inferential statistics, we need to imagine now that we don't have (or anyway cannot measure) all the data of interest.

And this is, of course, the typical situation. Consider:

* A zoologist wanting to know the typical lifespan of a Siberian tiger
* A cosmologist wanting to know the mass of a normal white dwarf star
* A businesswoman wanting to know how many M&M's her customers should expect to find in their Party Size bags
* A botanist wanting to know how tall California redwoods usually grow

The zoologist could, in principle:

Keep track of every currently existing Siberian tiger;
record their (more or less) exact ages at their moments of death;
add up those ages and divide by the number of tigers to calculate an average lifespan
––But only in principle. In all of these situations, there is no realistic or practical opportunity to check each relevant data point.

What we can do, however, is to check some of the data points we want to check. That is, we'll draw a sample of data from our population of interest. We can then use the techniques of descriptive statistics to characterize our sample.

Does this help? The hope, of course, is that our sample will be representative of the population as a whole, which would justify our using facts about the sample to infer things about the population as a whole. But naturally we'll expect a certain amount of error: If I take the mean of a sample, $\bar{x}$ and project it as an estimate of the mean of the whole population, $\mu$, the estimate is bound to be imperfect.

Inferential statistics makes all this precise. Some of this gets fairly technical, hence the need for a whole unit on it!

**Quick discussion question:**

<details open>
<summary>Can anyone give an example of how inferential statistics is being used in the news today?</summary>
<br>
1. emergency preparedness<br/>
2. predicting illness/spread of infection <br/>
3. political campaigns
</details>

# Review Set Theory 

1. What is it? 
2. Why do we need it? 

**1. What is it?**<br/>
A set is a collection of some items (elements). We often use capital letters to denote a set. To define a set we can simply list all the elements in curly brackets. 

**For example:** to define a set **A** that consists of the two elements ♣ and ♢, we write **A**={♣,♢}. To say that ♢ belongs to **A**, we write **♢∈A**, where "∈" is pronounced "belongs to." To say that an element does not belong to a set, we use ∉. For example, we may write **♡∉A**.

![](https://i.pinimg.com/originals/11/3d/46/113d4693d4edf453430927b4558f6640.jpg)

**The symbols "|" and ":" are pronounced "such that."**

A={x|x satisfies some property} <br/>
**or**<br/>
A={x:x satisfies some property}



**If the set C is defined as C={x|x∈ℤ,−2≤x<10}, then C={−2,−1,0,⋯,9}.**

**Inclusion-exclusion principle:**
Note that if you want to know how many elements are in set *A* versus set  *B*, you can't simply sum up the elements, because they have elements in common.

In combinational mathematics, the inclusion-exclusion principle is a counting technique that solves this problem.

When having two sets, the method for counting the number of elements in the union of two finite sets is given by:

|A∪B|=|A|+|B|−|A∩B|,

|A∪B∪C|=|A|+|B|+|C|−|A∩B|−|A∩C|−|B∩C|+|A∩B∩C|.


<details open>
    
![](https://upload.wikimedia.org/wikipedia/commons/c/c4/Inclusion-exclusion.png)
</details>

**2. Why do we need it?**<br/> 
In order to understand/calculate probability we need a unifying theory to describe the data or sets that we're analyzing. 

# Classical Probability 

In 1814, Pierre-Simon Laplace wrote:

> Probability theory is nothing but common sense reduced to calculation. ... [Probability] is thus simply a fraction whose numerator is the number of favorable cases and whose denominator is the number of all the cases possible ... when nothing leads us to expect that any one of these cases should occur more than any other.

*Pierre-Simon, marquis de Laplace was a French scholar and polymath whose work was important to the development of engineering, mathematics, statistics, physics, astronomy, and philosophy.*

### Terminology

**Outcome:** A result of a random experiment.<br/>
**Universal Set:** $\Omega$ <br/>
**Sample Space:** The set of all possible outcomes. <br/>
**Event:** A subset of the sample space.

**Axioms of Probability:**

**1:** For any event A, P(A)≥0.

**2:** Probability of the sample space S is P(S)=1.

**3:** If A1,A2,A3,⋯ are disjoint events, then P(A1∪A2∪A3⋯)=P(A1)+P(A2)+P(A3)+⋯

The basic idea is that if some events are disjoint (i.e., there is no overlap between them), then the probability of their union must be the summations of their probabilities.

# Handwritten example - axiom 3  
In a presidential election, there are four candidates. Call them A, B, C, and D. Based on our polling analysis, we estimate that A has a 20 percent chance of winning the election, while B has a 40 percent chance of winning. What is the probability that A or B win the election?

Notice that the events that {A wins}, {B wins}, {C wins}, and {D wins} are disjoint since more than one of them cannot occur at the same time. For example, if A wins, then B cannot win. From the third axiom of probability, the probability of the union of two disjoint events is the summation of individual probabilities. Therefore,

P(A wins or B wins)	=P({A wins}∪{B wins})<br/>
=P({A wins})+P({B wins})<br/>
=0.2+0.4<br/>
=0.6

*It is worth noting that we often write P(1) instead of P({1}) to simplify the notation, but we should emphasize that probability is defined for sets (events) not for individual outcomes. Thus, when we write P(2)=16, what we really mean is that P({2})=16.*

### Handwritten Example 2
1. There is a 60 percent chance that it will rain today.
2. There is a 50 percent chance that it will rain tomorrow.
3. There is a 30 percent chance that it does not rain either day.


<details open>
1. P(today)=0.6,<br/>
2. P(tomorrow)=0.5,<br/>
3. P(todayc∩tomorrowc)=0.3
</details>

### Python Example 

In [1]:
from fractions import Fraction

def P(event, space): 
    "The probability of an event, given a sample space."
    return Fraction(cases(favorable(event, space)), 
                    cases(space))

favorable = set.intersection # Outcomes that are in the event and in the sample space
cases     = len              # The number of cases is the length, or size, of a set

### Warm-up Problem: Die Roll
What's the probability of rolling an even number with a single six-sided fair die? Mathematicians traditionally use a single capital letter to denote a sample space; I'll use D for the die:

In [2]:
D     = {1, 2, 3, 4, 5, 6} # a sample space
even  = {   2,    4,    6} # an event

P(even, D)

Fraction(1, 2)

### Good. Let's do a few more

In [3]:
prime = {2, 3, 5, 7, 11, 13}
odd   = {1, 3, 5, 7, 9, 11, 13}

In [4]:
P(odd, D)

Fraction(1, 2)

In [5]:
P((even | prime), D) # The probability of an even or prime die roll

Fraction(5, 6)

In [6]:
P((odd & prime), D) # The probability of an odd prime die roll

Fraction(1, 3)

# Permutations
**An ordering of objects.** 
1. How many different ways can you could order the letters CAT?
$n!$
2. How many different ways can you order only 2 of the letters in CAT? 
$P_{k}^{n}= \dfrac{n!}{(n-k)!}$
3. What about repeats? How many different ways could we rewrite the letters in Mississippi? Divide out the duplicates. 



# Combinations
**Order is not important** 
$C_{k}^{n}= \dfrac{n!}{(n-k)!*k!}$

#### Examples 

**1.** If there are 10 people on a committee. How many ways are there of choosing a President, Vice President and Secretary? 

**2.** If there are 30 people on your tennis team, how many ways could you choose 2 co-captains? 

**3.** If a board has 10 women and 7 men, how many ways can you form a committee containing 4 members such that:<br/> 
    **a.** All 4 are men? <br/>
    **b.** 2 are men 2 are women?<br/>
    **c.** At least 1 is a woman? 

#### Card Problems

1. Count the number of possible five-card hands that can be dealt from a standard deck of 52 cards
2. Count the number of ways that a particular type of poker hand can occur
3. The probability of being dealt any particular type of hand is equal to the number of ways it can occur divided by the total number of possible five-card hands.

Consider dealing a hand of five playing cards. An individual card has a rank and suit, like 'J♥' for the Jack of Hearts, and a deck has 52 cards:<br/>
[Unicode documentation](https://www.linuxtopia.org/online_books/programming_books/introduction_to_python/python_tut_17.html)

In [7]:
suits = u'♥♠♦♣' 
ranks = u'AKQJT98765432'
deck  = [r + s for r in ranks for s in suits]
len(deck)

52

In [8]:
deck[0]

'A♥'

Now I want to define Hands as the sample space of all 5-card combinations from deck. The function *itertools.combinations* does most of the work; we than concatenate each combination into a space-separated string:

In [9]:
import itertools

def combos(items, n):
    "All combinations of n items; each combo as a space-separated str."
    return set(map(' '.join, itertools.combinations(items, n)))

Hands = combos(deck, 5)
len(Hands)

2598960

There are too many hands to look at them all but we can look at a sample.

In [10]:
import random
random.sample(Hands, 7)

['Q♠ 9♥ 9♦ 4♠ 3♣',
 'A♣ Q♣ 9♥ 5♣ 3♠',
 'T♦ 8♠ 5♠ 4♥ 3♥',
 '9♦ 8♣ 7♠ 7♣ 3♥',
 'K♣ 8♥ 7♣ 6♣ 3♦',
 'K♥ T♥ 6♠ 6♣ 5♣',
 'Q♦ Q♣ 7♣ 4♣ 2♠']

In [11]:
random.sample(deck, 7)

['J♣', 'A♦', '2♠', 'T♣', '7♣', '2♣', 'A♥']

Now we can answer questions like what's the probability of being dealt a flush (5 cards of the same suit) **:**

1. 52C5 = 52! / 5!(52 - 5)! = 52! / 5!47! = 2,598,960
2. Next, count the number of ways that five cards from a 52-card deck can be arranged to produce a flush. It requires two independent choices to produce a flush:

- Choose the rank of each card in the hand. A playing card can have a rank of 2, 3, 4, 5, 6, 7, 8, 9, 10, jack, queen, king, or ace. That's 13 distinct ranks. So, we choose five ranks from a set of 13 ranks. The number of ways to do this is 13C5.

- Choose one suit for the hand. There are four suits, from which we choose one. The number of ways to do this is 4C1.

- The number of ways to produce a flush is equal to the product of the number of ways to make each independent choice. Therefore,

flush = 13C5 * 4C1

flush = 1,287 * 4 = 5,148

Now, what do we need to do to get the probability of getting a flush? 

In [12]:
flush = {hand for hand in Hands if any(hand.count(suit) == 5 for suit in suits)}

P(flush, Hands)

Fraction(33, 16660)